### PR TITLE
Add more tests for doc comment fix and fix an edge case.

### PR DESCRIPTION
**Pull request by munificent**
A line that didn't have a leading `*` but did have a `*` elsewhere in
the line would have everything before the `*` trimmed.

cc @lrhn

### CODE REVIEW

File: test/fixes/doc_comments.stmt [hunk 1]
```diff
@@ -110,4 +110,50 @@ m() {}
...
+ * ```
+*/
+m() {}
+<<<
+/// Floo the grumshack.
+///
+/// Example:
+/// ```dart
+/// /** Doc comment */
+/// var grumshack = getGrumshack();
```
Comment (natebosch) [inline test/fixes/doc_comments.stmt]: https://www.dartlang.org/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

😝
